[Chua Kai Jun] iP

nujiak opened the pull request:
# Duke Upgraded 🤖
> The **best** thing to ever happen
> \- me, just now

*Duke Upgraded* helps you with remembering your everyday things:

* To-Dos
* Deadlines
* Events

## Getting Started

1. 🌐 Download the executable JAR from [Releases](https://github.com/nujiak/ip/releases). 
2. 🖱️ Double-click it.
3. 📝 Add your tasks.
4. 🤖 Let it manage your tasks for you!

## Features
- [x] Adding tasks
- [x] Deleting tasks
- [ ] Modifying tasks

## Preview
Here is a preview of the `main` method.

```java
public static void main(String[] args) {  
    Duke duke = new Duke("data/tasks.txt");  
  duke.run();  
}
```

Commit:
Add Gradle support

nujiak committed:
Implemented a skeletal version of Duke that greets and echoes the user.

nujiak committed:
Added ability to store and display text entered by user.

nujiak committed:
Added ability to mark tasks as done.

nujiak committed:
Add support for tracking ToDos, Deadlines, and Events.

nujiak committed:
Implemented automated test UI testing using I/O redirection technique.

nujiak committed:
Added error handling to handle invalid user inputs.

nujiak committed:
Added support for deleting tasks from the list.

nujiak committed:
Fixed comments for regex patterns.

nujiak committed:
Add DukeMemory class to save and restore tasks.

nujiak committed:
Change date in Deadline and Event to LocalDate.

nujiak committed:
Merge branch 'branch-Level-7'

nujiak committed:
Merge branch 'branch-Level-8'

nujiak committed:
Fix issues from merging branch-Level-8.

nujiak committed:
Add TaskList to manage tasks.

nujiak committed:
Add UI to handle user interface.

nujiak committed:
Add Parser and Command to handle input parsing.

nujiak committed:
Organize classes into packages.

nujiak committed:
Add JUnit unit tests.

nujiak committed:
Add manifest.

nujiak committed:
Add javadoc comments.

nujiak committed:
Modify code to comply with course coding standard.

nujiak committed:
Add find command to search for tasks.

nujiak committed:
Merge branch 'branch-A-CodingStandard'

nujiak committed:
Merge branch 'branch-A-Javadoc'

# Conflicts:
#	src/main/java/duke/command/AddDeadlineCommand.java
#	src/main/java/duke/command/AddEventCommand.java
#	src/main/java/duke/command/AddTodoCommand.java
#	src/main/java/duke/command/DeleteCommand.java
#	src/main/java/duke/command/MarkAsDoneCommand.java

nujiak committed:
Merge branch 'A-Level-9'

# Conflicts:
#	src/main/java/duke/Ui.java

kwokieee commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,37 @@
+     * @param input single-line user input
+     * @return duke.commands.Command which performs the corresponding action.
+     */
+    static Command parse(String input) {
```
I like how your parse method is easy to read and understand!

kwokieee commented on src/main/java/duke/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,45 @@
+    /**
+     * Regex pattern for finding deadline commands
+     */
+    private static final Pattern PATTERN_DEADLINE = Pattern.compile("^deadline (.*) /by (\\d{4}-\\d{2}-\\d{2})$");
```
```suggestion
    private static final Pattern DEADLINE_PATTERN = Pattern.compile("^deadline (.*) /by (\\d{4}-\\d{2}-\\d{2})$");
```

kwokieee commented on src/main/java/duke/command/AddEventCommand.java:
```diff
@@ -0,0 +1,43 @@
+    /**
+     * Regex pattern for finding event commands
+     */
+    private static final Pattern PATTERN_EVENT = Pattern.compile("^event (.*) /at (\\d{4}-\\d{2}-\\d{2})$");
```
```suggestion
    private static final Pattern EVENT_PATTERN = Pattern.compile("^event (.*) /at (\\d{4}-\\d{2}-\\d{2})$");
```

kwokieee commented on src/main/java/duke/command/AddTodoCommand.java:
```diff
@@ -0,0 +1,43 @@
+    /**
+     * Regex pattern for finding todo commands
+     */
+    private static final Pattern PATTERN_TODO = Pattern.compile("^todo (.*)$");
```
```suggestion
    private static final Pattern TODO_PATTERN = Pattern.compile("^todo (.*)$");
```

kwokieee reviewed (commented):
LGTM, nice job on giving names that are easy to read and understandable! 😊 I have indicated a few minor suggestions on a few names.

nujiak committed:
Merge branch 'add-gradle-support' of https://github.com/nujiak/ip

nujiak committed:
Add checkstyle.xml and suppressions.xml for gradle checkstyle tasks.

nujiak committed:
Add GUI to Duke using JavaFX.

nujiak committed:
Merge branch 'branch-Level-10'

junwei26 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,37 @@
+package duke;
+
+import duke.command.*;
```
You might want to import the classes individually instead of importing all at once

junwei26 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,89 @@
+     * Greets the user when Duke is started.
+     */
+    public static String greet() {
+        return "Hello, I'm Duke! I'm here to help you with ANYTHING";
```
You might want to declare this string as a constant so that it is easier to edit or change in future

junwei26 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,89 @@
+     * Used after the bye command.
+     */
+    public static String goodbye() {
+        return "Bye bye, see you next time.";
```
Same for this string too, might want to change it to a constant too

junwei26 reviewed (commented):
Great job! There wasn't any major code standard violation. Sorry for nitpicking as I really couldn't find any obvious violation

nujiak committed:
Fix IndexOutOfBoundsException when deleting.

nujiak committed:
Add FXML to separate code and UI.

nujiak committed:
Merge branch 'branch-gui-design'

nujiak committed:
Change mainClassName for Gradle tasks showJar and application.

nujiak committed:
Add assertations.

Some methods do not accept null as parameters.

Let's use assertations to ensure that these methods do not receive
null as parameters.

If they do, the program will be stopped and this issue can be
explored by reading the stacktrace from the AssertationError thrown.

nujiak committed:
Enable CheckStyle.

Certain parts of the code do not abide by the Coding Standard of the
module.

Let's enable CheckStyle which will flag any violation of the Coding
Standard so they can be fixed.

nujiak committed:
Fix all violations of the module's Coding Standard.

nujiak committed:
Merge pull request #1 from nujiak/branch-A-Assertations

Add assertations.

nujiak committed:
Merge pull request #2 from nujiak/branch-A-CodeQuality

Improve Code Quality

nujiak committed:
Add sort command for sorting tasks alphabetically or
chronologically.

nujiak committed:
Merge branch 'branch-C-Sort'

nujiak committed:
Merge branch 'master' into branch-gui-design

nujiak committed:
Add window title.

nujiak committed:
Add initial message from Dude.

nujiak committed:
Merge branch 'branch-gui-design'

nujiak committed:
Add Ui.png for product website.

nujiak committed:
Set theme jekyll-theme-merlot

nujiak committed:
Update readme with basic user guide.

nujiak committed:
Merge branch 'master' of https://github.com/nujiak/ip
